controllers/openstackdataplane_controller.go

```diff
@@ -135,7 +142,15 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	var deployErrors []string
 	shouldRequeue := false
-	if instance.Spec.DeployStrategy.Deploy {
```

Comment:
we might want to keep this guard as well, otherwise the whole DeployStrategy is a subject of removal within this change

Reply:
It probably doesn't make sense to keep it tbh. If a user creates or updates the Dataplane spec, and removes the annotation. They would expect it to deploy like any other Kubernetes resource that has had it's spec updated. Having this as part of the spec doesn't align well with Kubernetes constructs and duplicates the intention behind the annotation I guess.

I think we're better off just removing it. But there are a lot of places I need to do that, and they each require some careful consideration around the original intention, and how that intention can be preserved as part of this change. Leave it with me, I'll send another revision of this PR early next week when I'm back from leave and remove these. I'll be interested to hear your inputs on my implementation to see what you think about it.